### PR TITLE
DPGO-4. Doctor CRUD.

### PR DESCRIPTION
**Pull request by KekemonBS**
Created doctor CRUD.

Changed entites, handlers, usecases and router.go.

### CODE REVIEW

File: internal/repository/postgres/doctor/repo.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	return nil
+}
+
+// Update updates a doctor
+func (r *Repository) Update(ctx context.Context, d *entity.Doctor) error {
+	query := `UPDATE doctors SET id = $1, first_name = $2, 
+			  last_name = $3, speciality = $4, start_at = $5, end_at = $6 `
+	res, err := r.storage.ExecContext(ctx,
+		query,
+		&d.ID,
```
Comment (glbter) [inline internal/repository/postgres/doctor/repo.go]: looks like an error, the id should not change

File: internal/entity/doctor.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import "time"
+
+// Doctor struct (Model) and request
+type Doctor struct {
+	ID         string    `json:"id" validate:"required"`
+	FirstName  string    `json:"name" validate:"required"`
+	LastName   string    `json:"last_name" validate:"omitempty"`
+	Speciality string    `json:"speciality" validate:"required"`
+	StartAt    time.Time `json:"start_at"`
```
Comment (glbter) [inline internal/entity/doctor.go:11]: I can't get it. For now we are assuming, that the doctor has a fixed schedule?

File: internal/server/http/handlers/doctor/handlers.go [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+// NewHandlers create new doctor handlers.
+func NewHandlers(uc DoctorsUsecases, log *zap.Logger) *Handlers {
+	return &Handlers{usecases: uc, logger: log}
+}
+
+// GetDoctors Get all doctors.
+func (h *Handlers) GetDoctors(w http.ResponseWriter, r *http.Request) {
+	doctors, err := h.usecases.GetAll(r.Context())
+	if err != nil {
+		h.writeErrorResponse(http.StatusInternalServerError, err.Error(), w)
```
Comment (glbter) [inline internal/server/http/handlers/doctor/handlers.go:44]: please, don't expose internal error to users

File: internal/server/http/handlers/doctor/handlers.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+		return
+	}
+	d.ID = id
+	if ok := isRequestValid(&d); !ok {
+		h.writeErrorResponse(http.StatusBadRequest, "doctor data invalid", w)
+		return
+	}
+	if err := h.usecases.Update(r.Context(), &d); err != nil {
+		h.logger.Error("can't update a doctor", zap.Error(err))
+		if errors.Is(err, customerrors.NotFound) {
```
Comment (glbter) [inline internal/server/http/handlers/doctor/handlers.go:104]: you could check `error.Is` before `err != nil` to remove nesting

**Review by glbter — CHANGES_REQUESTED** [review]
well done!!!

File: internal/entity/doctor.go (hunk 2 above)
Comment (KekemonBS) [reply internal/entity/doctor.go:11]: yes

**Review by glbter — APPROVED** [review]

**Pull request merged**